### ISSUE TITLE
Make _Groupby.apply take additional arguments. 

### ISSUE THREAD

**Issue by gabrielelanaro** [dask/dask#3234]
In pandas API we have that apply takes extra *args and **kwargs https://pandas.pydata.org/pandas-docs/stable/generated/pandas.core.groupby.GroupBy.apply.html#pandas.core.groupby.GroupBy.apply

but in dask this doesn't happen:
https://github.com/dask/dask/blob/master/dask/dataframe/groupby.py#L1035

This is important because if you use apply with a closure with large arguments (using partial), there's no way to pass a future instead to save in memory and serialization costs.

**Comment by gabrielelanaro** [comment]
I can make a PR for this, but I see there could be also some other issue (dask 0.17.1) (distributed 1.21.0):

```
import pandas as pd
import numpy as np
from dask.distributed import Client, LocalCluster
import dask.dataframe as ddf

cluster = LocalCluster()
client = Client(cluster)

df = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
data = ddf.from_pandas(df, npartitions=4)
a_fut = client.scatter(1, broadcast=True)

def operate(df, a):
    return df.groupby(df.index).apply(lambda x, a: x + a, a)

data.map_partitions(operate, a_fut).compute()
# TypeError: cannot convert Future to an intblock
```

**Comment by mrocklin** [comment]
Currently the second argument for apply is meta=, which has a different
meaning.  You would need to implement the changes suggested above for this
to work.

**Comment by gabrielelanaro** [comment]
what I mean is that there's a problem in map_partitions itself. In the sense that a_fut doesn't get resolved:

```
def operate(df, a):
    print(df)
    print(a)


data.map_partitions(operate, a_fut).compute()
```
   A  B  C  D
0  1  1  1  1
1  1  1  1  1
<Future: status: finished, type: int, key: int-c0a8a20f903a4915b94db8de3ea63195>
```

**Comment by mrocklin** [comment]
I recommend raising a separate issue with a minimal example

**Comment by jcrist** [comment]
Fixed by #3256.
